stdlib/EvaluatorSettings.pkl

```diff
@@ -0,0 +1,125 @@
+
+/// The environment variables available to Pkl, read using the `env:` scheme.
+///
+/// Example:
+/// ```
+/// env {
+///   ["IS_PROD"] = "true"
+/// }
+/// ```
+env: Mapping<String, String>?
+
+/// The set of module URI patterns that can be imported.
+///
+/// Each element is a regular expression pattern that is tested against a module import.
+///
+/// Modules are imported either through an amends or extends clause, an import clause, or an
+/// import expression.
+///
+/// Example:
+/// ```
+/// allowedModules {
+///   "https:"
+///   "file:"
+///   "package:"
+///   "projectpackage:"
+/// }
+/// ```
+allowedModules: Listing<String(isRegex)>?
+
+/// The set of resource URI patterns that can be imported.
+///
+/// Each element is a regular expression pattern that is tested against a resource read.
+///
+/// Example:
+/// ```
+/// allowedResources {
+///   "https:"
+///   "file:"
+///   "package:"
+///   "projectpackage:"
+///   "env:"
+///   "prop:"
+/// }
+/// ```
+allowedResources: Listing<String(isRegex)>?
+
+/// Disables the file system cache for `package:` modules.
+///
+/// When caching is disabled, packages are loaded over the network and stored in memory.
+noCache: Boolean?
+
+/// A collection of jars, zips, or directories to be placed into the module path.
+///
+/// Module path modules and resources may be read and imported using the `modulepath:` scheme.
+///
+/// Relative paths are resolved against PklProject's enclosing directory.
+modulePath: Listing<String>?
+
+/// The duration after which evaluation of a source module will be timed out.
+///
+/// Note that a timeout is treated the same as a program error in that any subsequent source modules will not be evaluated.
+timeout: Duration?
+
+/// The directory where `package:` modules are cached.
+///
+/// Relative paths are resolved against PklProject's enclosing directory.
+moduleCacheDir: String?
+
+/// Restricts access to file-based modules and resources to those located under this directory.
+///
+/// Relative paths are resolved against PklProject's enclosing directory.
+rootDir: String?
+
+/// Configuration of outgoing HTTP requests.
+http: Http?
+
+/// Settings that control how Pkl talks to HTTP(S) servers.
+class Http {
+  /// Configuration of the HTTP proxy to use, `null` for none.
+  proxy: Proxy?
+}
+
+/// Settings that control how Pkl talks to HTTP proxies.
+class Proxy {
+  /// The proxy to use for HTTP(S) connections.
+  ///
+  /// At the moment, only HTTP proxies are supported.
+  ///
+  /// Example:
+  /// ```
+  /// local proxyPassword = read("my-proxy-password.txt").text
+  /// address = "http://myuser@\(proxyPassword):my.proxy.example.com:5080"
+  /// ```
+  address: Uri(startsWith("http://"))
+
+  /// Hosts to which all connections should bypass a proxy.
+  ///
+  /// Values can be either hostnames, or CIDR blocks.
+  ///
+  /// Follows the same rules as cURL's [--noproxy](https://curl.se/docs/manpage.html#--noproxy)
+  /// flag.
```

Comment:
I'm not totally sure that we do follow cURL's rules. Better to spell this all out:

```suggestion
  /// Values can be either hostnames, or IP addresses.
  /// IP addresses can optionally be provided using [CIDR notation](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing#CIDR_notation).
  ///
  /// The only wildcard is `"*"`, which disables all proxying.
  ///
  /// A hostname will match all subdomains.
  /// For example, `example.com` will match `foo.example.com`, but not `fooexample.com`.
  /// A hostname that is prefixed with a dot will match the hostname itself.
  /// Effectively, `.example.com` is the same as `example.com`.
  ///
  /// Optionally, a port can be specified.
  /// If a port is omitted, all ports are matched.
  ///
  /// Example:
  ///
  /// ```
  /// noProxy {
  ///   "127.0.0.1"
  ///   "169.254/16"
  ///   "example.com"
  ///   "localhost:5050"
  /// }
  /// ```
```